presto-iceberg/src/main/java/com/facebook/presto/iceberg/IcebergHiveMetadata.java

```diff
@@ -422,8 +443,8 @@ public List<SchemaTableName> listViews(ConnectorSession session, Optional<String
     {
         ImmutableList.Builder<SchemaTableName> tableNames = ImmutableList.builder();
         MetastoreContext metastoreContext = getMetastoreContext(session);
-        for (String schema : listSchemas(session, schemaName.orElse(null))) {
-            for (String tableName : metastore.getAllViews(metastoreContext, schema).orElse(emptyList())) {
+        for (String schema : listSchemas(session, Optional.ofNullable(schemaName.get()).orElse(null))) {
```

Comment:
If `schemaName` is already an optional, you don't need to wrap this with `Optional.ofNullable`. Just `schemaName.orElse(null)`